scylla-server/src/proxy/proxy-client.ts

```diff
@@ -39,26 +61,108 @@ export default class ProxyClient {
    */
   private handleOpen = (packet: IConnackPacket) => {
     console.log('Connected to Siren', packet.properties);
+    this.createNewRun = true;
     this.subscribeToTopics(Object.values(Topic));
   };
 
   /**
    * Handles messages received from Siren
+   * Parses as ServerMessage, otherwise throws error
+   * Passes data to handleData
    * @param topic The topic the message was received on
    * @param message The message received from Siren
    */
-  private handleMessage = (topic: string, payload: Buffer) => {};
-
+  private handleMessage = (topic: string, payload: Buffer) => {
+    // am i supposed to do something with topic
+    console.log('Received Message: ', payload);
+    try {
+      const data = JSON.parse(payload.toString()) as ServerMessage;
+      this.handleData(data);
```

Comment:
so my one issue with this right now is that if one part of handleData fails, the whole thing fails, and all the code is built upon that assumption, I think ideally we should future proof this a bit, because eventually we will wanna be able to salvage parts of a message even if other parts of it are wrong

Reply:
for being able to salvage parts of the message, what if we changed handleData so that it was just the stuff inside the for loop and then pulled out the rest into handleMessage, then handleMessage would iterate through the ServerData and call handleData in a try catch for each ServerData. That way if handleData fails for one ServerData, the whole thing doesn't fail.